### PR TITLE
Update grammar to aya 0.22

### PR DESCRIPTION
**Pull request by imkiva**
- Update grammar to aya 0.22
- Reuse `AyaGKProducer` to speed up compilation

### CODE REVIEW

File: src/main/grammar/AyaPsiParser.bnf [hunk 1]
```diff
@@ -250,7 +264,15 @@ sigmaExpr ::= KW_SIGMA tele+ SUCHTHAT expr
 lambdaExpr ::= KW_LAMBDA lambdaTele+ (IMPLIES expr)?
 matchExpr ::= KW_MATCH exprList clauses
 doExpr ::= KW_DO LBRACE? doBlock RBRACE?
+thisExpr ::= KW_THIS (AT qualifiedId)?
 idiomExpr ::= LIDIOM idiomBlock? RIDIOM
+pathExpr ::= LPATH pathTele+ RPATH expr? partialExpr?
+
+partialExpr ::= LPARTIAL (bareSubSystem barredSubSystem*)? RPARTIAL
+subSystem ::= expr DEFINE_AS expr
+bareSubSystem ::= subSystem
```
Comment (ice1000) [inline src/main/grammar/AyaPsiParser.bnf:273]: `private`

File: src/main/grammar/AyaPsiParser.bnf [hunk 2]
```diff
@@ -250,7 +264,15 @@ sigmaExpr ::= KW_SIGMA tele+ SUCHTHAT expr
...
+barredSubSystem ::= BAR subSystem
```
Comment (ice1000) [inline src/main/grammar/AyaPsiParser.bnf:274]: private

File: src/main/grammar/AyaPsiParser.bnf [hunk 3]
```diff
@@ -352,13 +374,13 @@ lambdaTeleBinder ::= teleBinderTyped
                    | lambdaTeleBinderAnonymous
 lambdaTeleBinderAnonymous ::= teleParamName
 
+// path type tele
+pathTele ::= teleParamName
```
Comment (ice1000) [inline src/main/grammar/AyaPsiParser.bnf:378]: Maybe `private`?

**Review by ice1000 — COMMENTED** [review]
:)

**Comment by imkiva** [comment]
bors merge

**Pull request merged**